### PR TITLE
Add basic support for radial gradients

### LINKED ISSUE #555: Add support for radial gradients
This should involve adding a radio toggle input to the Gradient tool options bar, the Properties Panel for shapes with gradients, and updating the SVG string generator to support radial gradients.

### PR DESCRIPTION
**Pull request by hannahli2010**
Add support for radial gradients 
- Add a radio toggle input to the Gradient tool options bar and the Properties Panel for shapes with gradients
- Update the SVG string generator to support radial gradients.
- Click and drag defines the centre of the radial gradient and the radius respectively

Closes #555

### CODE REVIEW

File: editor/src/document/properties_panel_message_handler.rs
Comment (0HyperCube) [inline editor/src/document/properties_panel_message_handler.rs:840]: I think `_ => unimplemented!()` is cleaner

File: graphene/src/layers/style/mod.rs
Comment (0HyperCube) [inline graphene/src/layers/style/mod.rs:64]: Perhaps it would be better to store the gradient type (currently in `gradient_tool`) as a property of the `Gradient` in the style?

**Review by 0HyperCube — APPROVED** [review]
Looks good.

File: graphene/src/layers/style/mod.rs
Comment (hannahli2010) [reply graphene/src/layers/style/mod.rs:64]: That sounds like a good idea! I'll make that change and also move the `GradientType` definition to this file instead since I think that would make more sense.

Comment (RobNadal) [reply graphene/src/layers/style/mod.rs:79]: @Hypercube0 we tried adding GradientType as a property to the Gradient struct and a default for GradientType (set to Linear). Despite this, we saw the below error pop up on the Graphite UI while testing our changes, which seems to imply the default isn't being applied to Gradient structs loaded up from existing projects. We're not sure if this is an issue from the Default, Serialize, or Deserialize traits. 
image

Comment (0HyperCube) [reply graphene/src/layers/style/mod.rs:79]: @RobNadal This is due to the autosaved document failing to load (due to the json format being changed). Feel free to ignore this.

Comment (Keavon) [reply graphene/src/layers/style/mod.rs]: @RobNadal for future reference, that happens commonly when the document serialization format changes from what is saved in your browser between page loads by the auto-save system. You just need to close the open document tab and it'll add a new blank tab that follows the new document format. Nothing to worry about!

File: graphene/src/layers/style/mod.rs
Comment (hannahli2010) [inline graphene/src/layers/style/mod.rs:132]: Since we are storing the linear vs radial type in the `Gradient`, it was unnecessary to differentiate between the two here. We're open to better naming conventions!

Comment (hannahli2010) [reply graphene/src/layers/style/mod.rs:132]: Resolving for now and rewriting this comment after the git rebase

Comment (hannahli2010) [inline graphene/src/layers/style/mod.rs:132]: Since we are storing the linear vs radial type in the `Gradient`, it was unnecessary to differentiate between the two here. We're open to better naming conventions!

**Review by 0HyperCube — APPROVED** [review]
Looks good

**Comment by otdavies** [comment]
You should be good to merge into master once the checks pass. Changes look good and @0HyperCube approved, nicely done.

**Comment by Keavon** [comment]
A little UI/UX feedback before merging.

 In the Properties panel:
image

All widgets should be on the right side of a label. "Type" or "Spread" would be reasonable name choices. (@otdavies do you have a preference between the two names, or other suggestions?)

**Comment by otdavies** [comment]
> All widgets should be on the right side of a label. "Type" or "Spread" would be reasonable name choices. (@otdavies do you have a preference between the two names, or other suggestions?)

I think that **Gradient Type** or just **Type** would be a descriptive label.

**Comment by Keavon** [comment]
Let's call it Gradient Type if there's room for it to fit. If not, then just Type.

**Comment by otdavies** [comment]
@hannahli2010 Noticed a small issue with the gradient tool swapping between linear / radial unexpectedly.

Reproducible Steps:
1. Draw shape
2. Add linear gradient to shape
3. Set gradient to radial
4. Drag one of the gradient handles
5. Issue will present, it will swap the gradient type unexpectedly.

I assume this is unintentional, just another of the many of cases to keep an eye out for. 😄

**Comment by Keavon** [comment]
Specifically, it reverts to what the tool is set to. So it does it from linear to radial also.

**Review by otdavies — APPROVED** [review]
UI changes look good and the gradient switch bug is resolved. I think this is good to go. You agree @Keavon?

**Pull request merged**
